packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -237,6 +241,7 @@ function useBlockEditorSettings( settings, postType, postId ) {
 					BLOCK_EDITOR_SETTINGS.includes( key )
 				)
 			),
+			__experimentalStyles: styles,
```

Comment:
The settings already contain a `styles` key (it's in the `settings` object [above](https://github.com/WordPress/gutenberg/pull/59929/files#diff-278a4a275170c4215a2f4e9d5c0c6d831f9328258220a76cda04b8d9f39f01a5R240))

This `styles` object contains the "compiled" styles, (the CSS) which is not what we want here. What we want is the styles object that contains the user styles.

Another alternative to adding a new `__experimentalStyles` key might be to nest both the "compiled" and "uncompiled" styles under `settings.styles`:

- `settings.styles.raw` - the "uncompiled" styles (what we're adding here)
- `settings.styles.css` - the styles currently present under `settings.styles`

The naming is of course debatable, I'm only suggesting what the structure might be.

Reply:
Yes, I think the main blocker of this PR is the naming. It's also important in the discussion to consider that we also have `__experimentalFeatures` that corresponds to the "settings" part of global styles and this new key corresponds to the "styles" part.

The other thing to note is that some of the "compiled" styles mention above are a duplication of the "raw" styles but not all of them. Some other styles come from CSS files themes or plugins inject.

Reply:
@WordPress/gutenberg-core We need your help coming up with the best API here for the block editor package (generic package to build block editors). How should we pass the "global styles" styles and settings and what about "extra styles" that can be provided as raw CSS to be applied to the editor canvas.

Reply:
At 1st sight I like the idea to split styles in two:

```
styles.source
styles.compiled
```

I think the concept of `raw` doesn't apply since we usually take something raw and transform it into a prepared version. In this case tho we compile to CSS some JSON data.

Reply:
> What we want is the styles object that contains the user styles.

Why can't we then call the key `userStyles`?

Reply:
We probably can! I am not yet 100% sure that we don't need ALL the styles in the near future (theme and user-defined ones) as I discover what's needed for the implementation. That's the only reason why I didn't name it `userStyles` 🙂

Reply:
Noting that we can't split the existing "styles" setting for backward compatibility reasons.

Reply:
> Noting that we can't split the existing "styles" setting for backward compatibility reasons.

It would be neat to "reclaim" styles for the global styles object, either by splitting as mentioned above or otherwise. 

Blocks use "styles" for the style object and so does theme.json. 

Maybe putting compiled CSS under `"css"` 🤷🏻 

Could we do some sort of type check for backwards compat? E.g.,

```js
// Assume current structure of `[ { ... }, { ... } ]` 
// If we want to be sure, `styles?.[0].hasProperty( isGlobalStyles )`
if ( Array.isArray( styles ) ) { ... }
```

Reply:
Maybe rename this new attribute to `globalStyles` and rename the existing `styles` attribute to `css`?

Reply:
> rename the existing styles attribute to css?

Wouldn't that break back compat?

Reply:
+1 for `globalStyles` 

It covers the potential to include theme and user-defined styles as noted above.

Reply:
> > rename the existing styles attribute to css?
> 
> Wouldn't that break back compat?

I should have specified "rename in a backwards compatible way" :) so support both but prefer `css`.

Reply:
Good discussion!

+1 for renaming/repurposing styles - globalStyles is as good as any name. Will there ever be `customStyles` or `userStyles`, or do we assume that global styles will be theme + user merged?

Even if it's noisy for a while to rename/deprecate, I don't think backwards compatibility should be so stultifying a mandate that it should get in the way of progress! Onwards!